plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/protobuf/TestKafkaProtobufWithSchemaRegistryMinimalFunctionality.java

```diff
@@ -170,10 +168,13 @@ public void testUnsupportedRecursiveDataTypes()
                 messages.stream(),
                 producerProperties());
 
-        // Any operation which performs schema parsing leads to failure so we cannot use #waitUntilTableExists
-        assertQueryFailsEventually("SELECT * FROM " + toDoubleQuoted(topic),
-                "\\Qstatement is too large (stack overflow during analysis)\\E",
-                succinctDuration(2, SECONDS));
+        waitUntilTableExists(topic);
+        assertQueryFails("SELECT * FROM " + toDoubleQuoted(topic),
+                "Protobuf schema containing fields with self-reference are not supported because they cannot be mapped to a Trino type: " +
+                        "io.trino.protobuf.RecursiveValue > " +
```

Comment:
Is `a > b > c ...` supposed to indicate the hierarchy of reference? Asking because `Set<String> processedMessages` doesn't guarantee ordering.

Reply:
The concrete type is ImmutableSet which guarantees stable iteration order (insertion order). However I'll see if we can avoid depending on that detail.